tools/reactive-controllers/test/pending-state.test.ts

```diff
@@ -0,0 +1,122 @@
+/*
+Copyright 2020 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+/*
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import { expect, fixture, html } from '@open-wc/testing';
+import { ReactiveElement } from 'lit';
+import { PendingStateController } from '@spectrum-web-components/reactive-controllers/src/PendingState.js';
+import sinon from 'sinon';
+import '@spectrum-web-components/progress-circle/sp-progress-circle.js';
+import '@spectrum-web-components/picker/sp-picker.js';
+
+describe('PendingStateController', () => {
+    let host: ReactiveElement;
+    let pending: sinon.SinonStub;
+    let onPendingChange: sinon.SinonStub;
+    let controller: PendingStateController<ReactiveElement>;
+
+    beforeEach(async () => {
+        host = await fixture<ReactiveElement>(html`
+            <sp-picker pending></sp-picker>
+        `);
+        pending = sinon.stub();
+        onPendingChange = sinon.stub();
+        controller = new PendingStateController(host, {
+            pending,
+            onPendingChange,
+        });
+    });
+
+    afterEach(() => {
+        sinon.restore();
+    });
+
+    describe('isPending', () => {
+        it('should call the pending function', () => {
+            controller.isPending();
+            expect(pending).to.have.been.called;
+        });
```

Comment:
Can you add a test to toggle the `pending` state on and off (atleast once) to verify if the component state is being preserved correctly.

Reply:
Done!